sqlmesh/core/engine_adapter/redshift.py

```diff
@@ -55,15 +55,57 @@ class RedshiftEngineAdapter(
         },
     )
 
-    def _columns_query(self, table: exp.Table) -> exp.Select:
+    def columns(
+        self,
+        table_name: TableName,
+        include_pseudo_columns: bool = True,
+    ) -> t.Dict[str, exp.DataType]:
+        table = exp.to_table(table_name)
+
         sql = (
-            exp.select("column_name", "data_type")
+            exp.select(
+                "column_name",
+                "data_type",
+                "character_maximum_length",
+                "numeric_precision",
+                "numeric_scale",
+            )
             .from_("svv_columns")  # Includes late-binding views
             .where(exp.column("table_name").eq(table.alias_or_name))
         )
         if table.args.get("db"):
             sql = sql.where(exp.column("table_schema").eq(table.args["db"].name))
-        return sql
+
+        columns_raw = self.fetchall(sql, quote_identifiers=True)
+
+        def build_var_length_col(row: tuple) -> tuple:
+            var_len_chars = (
```

Comment:
make this a set and put it outside of the function since this is done in a tight loop